test/test_buffer_protocol.py

```diff
@@ -0,0 +1,175 @@
+
+    @onlyCPU
+    @dtypes(*common.torch_to_numpy_dtype_dict.keys())
+    def test_with_offset(self, device, dtype):
+        # Offset should be valid whenever there is, at least,
+        # one remaining element
+        for i in range(SIZE):
+            self._run_test(SHAPE, dtype, first=i)
+
+    @onlyCPU
+    @dtypes(*common.torch_to_numpy_dtype_dict.keys())
+    def test_with_count(self, device, dtype):
+        # Count should be valid for any valid in the interval
+        # [-1, len(input)], except for 0
+        for i in range(-1, SIZE + 1):
+            if i != 0:
+                self._run_test(SHAPE, dtype, count=i)
+
+    @onlyCPU
+    @dtypes(*common.torch_to_numpy_dtype_dict.keys())
+    def test_with_count_and_offset(self, device, dtype):
+        # Explicit default count [-1, 1, 2, ..., len]
+        for i in range(-1, SIZE + 1):
+            if i != 0:
+                self._run_test(SHAPE, dtype, count=i)
+        # Explicit default offset [0, 1, ..., len - 1]
+        for i in range(SIZE):
+            self._run_test(SHAPE, dtype, first=i)
+        # All possible combinations of count and dtype aligned
+        # offset for 'input'
+        # count:[1, 2, ..., len - 1] x first:[0, 1, ..., len - count]
+        for i in range(1, SIZE):
+            for j in range(SIZE - i + 1):
+                self._run_test(SHAPE, dtype, count=i, first=j)
+
+    @onlyCPU
+    @dtypes(*common.torch_to_numpy_dtype_dict.keys())
+    def test_invalid_positional_args(self, device, dtype):
+        bytes = get_dtype_size(dtype)
+        in_bytes = SIZE * bytes
+        # Empty array
+        with self.assertRaisesRegex(ValueError,
+                                    r"both buffer length \(0\) and count"):
+            empty = numpy.array([])
+            torch.frombuffer(empty)
+        # Count equals 0
+        with self.assertRaisesRegex(ValueError,
+                                    r"both buffer length .* and count \(0\)"):
+            self._run_test(SHAPE, dtype, count=0)
+        # Offset negative and bigger than total length
+        with self.assertRaisesRegex(ValueError,
+                                    rf"offset \(-{bytes} bytes\) must be"):
+            self._run_test(SHAPE, dtype, first=-1)
+        with self.assertRaisesRegex(ValueError,
+                                    rf"offset \({in_bytes} bytes\) must be .* "
+                                    rf"buffer length \({in_bytes} bytes\)"):
+            self._run_test(SHAPE, dtype, first=SIZE)
+        # Non-multiple offset with all elements
+        if bytes > 1:
+            offset = bytes - 1
+            with self.assertRaisesRegex(ValueError,
+                                        rf"buffer length \({in_bytes - offset} bytes\) after "
+                                        rf"offset \({offset} bytes\) must be"):
+                self._run_test(SHAPE, dtype, offset=bytes - 1)
+        # Count too big for each good first element
+        for first in range(SIZE):
```

Comment:
This is fine and doesn't need to change in this PR, but in the future it might be simpler to just try one invalid input of this type instead of enumerating every invalid input